controller/query.go

```diff
@@ -183,6 +184,52 @@ func (c *QueryController) Show(ctx *app.ShowQueryContext) error {
 	return ctx.OK(res)
 }
 
+// Update runs the update action.
+func (c *QueryController) Update(ctx *app.UpdateQueryContext) error {
+	currentUser, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+	if ctx.Payload.Data.Attributes.Version == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("data.attributes.version", nil).Expected("not nil"))
+	}
+	var q *query.Query
+	err = application.Transactional(c.db, func(appl application.Application) error {
+		var err error
+		q, err = appl.Queries().Load(ctx.Context, ctx.QueryID, ctx.SpaceID)
+		if err != nil {
+			return err
+		}
+		if q.Creator != *currentUser {
+			log.Warn(ctx, map[string]interface{}{
+				"query_id":     ctx.QueryID,
+				"creator":      q.Creator,
+				"current_user": *currentUser,
+			}, "user is not the query creator")
+			return errors.NewForbiddenError("user is not the query creator")
+		}
+		if q.Version != *ctx.Payload.Data.Attributes.Version {
+			return errors.NewVersionConflictError("version conflict")
+		}
+		if ctx.Payload.Data.Attributes.Title != "" {
+			q.Title = strings.TrimSpace(ctx.Payload.Data.Attributes.Title)
```

Comment:
@baijum sorry for the late review. I wonder if the `if` shouldn't look like this:

```go
if strings.TrimSpace(ctx.Payload.Data.Attributes.Title) != "" {
    q.Title = strings.TrimSpace(ctx.Payload.Data.Attributes.Title)
}
````

This will prevent people from updating the title to `"     "` for example.

Reply:
I think whitespace only title is not good.

Reply:
@baijum yes, that's my point ;). But your logic sets the title even if it is `"    "` and even worse is sets it to `""`. Not sure if that is intended.

Reply:
The actual validation happens in the repo/model. The logic here is intent to identify whether user has send some value or nor. If the title empty string, that indicates user has not send the value for update and the original one will be kept. The same thing for JSON field also.

Reply:
I see. But why check for `""` then? If a title is indeed empty and the repo catches that, then just let the repo do all the job, no?

Reply:
The model function expects the Query object with all values set. If the user input is non-empty string then we override the value and send it to model function.